docs/tutorial/alf_snapshot_and_advanced_play.rst

```diff
@@ -0,0 +1,249 @@
+repo*.
+
+When playing a trained model with snapshot enabled, we can add an option ``--use_alf_snapshot``
+to use the archived ALF version instead of the current one:
+
+.. code-block:: bash
+
+    python -m alf.bin.play --root_dir /tmp/alf_tutorial1 --use_alf_snapshot
+
+You'll notice some message like
+
+.. code-block:: bash
+
+    I0914 11:49:21.344848 140367115069248 play.py:168] === Using an ALF snapshot at '/tmp/alf_tutorial1/alf' ===
+
+which indicates that you're indeed using a snapshot version.
+
+.. note::
+
+    It's possible to turn off the snapshot feature. When training or grid searching,
+    appending the option ``--nostore_snapshot`` will do so.
+
+The special case of config file
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The config file of a training job will also be archived in a snapshot, if it's
+*located under ALF root*. However, ALF also archives another config copy
+directly under the job directory named as ``alf_config.py``. This special config
+file also records *command-line config parameters and grid search parameters*, so
+it's generally more accurate than the snapshot version.
+
+In the above training example, if we add a command-line config parameter:
+
+.. code-block:: bash
+
+    rm -rf /tmp/alf_tutorial1
+    python -m alf.bin.train --root_dir /tmp/alf_tutorial1 --conf <ALF_ROOT>/alf/examples/tutorial/minimal_example_conf.py --conf_param="TrainerConfig.summary_interval=100"
+
+Again, ALF will store a snapshot at ``/tmp/alf_tutorial1/alf`` and we can get
+the config file at
+
+.. code-block:: bash
+
+    /tmp/alf_tutorial1/alf/examples/tutorial/minimal_example_conf.py
+
+However, this is just a copy of the original config file: it *doesn't* record
+our command-line parameter ``TrainerConfig.summary_interval=100``.
+
+In contrast, if we look at ``/tmp/alf_tutorial1/alf_config.py``, we'll see something
+like
+
+.. code-block:: python
+
+    ########### pre-configs ###########
+
+    import alf
+    alf.pre_config({
+        'TrainerConfig.summary_interval': 100,
+    })
+
+    ########### end pre-configs ###########
+
+on the very top of the file.
+
+Regardless of whether having the flag ``--use_alf_snapshot`` when playing a model,
+ALF will always use ``alf_config.py``. So if we'd like to make changes to the
+config file for play, we need to modify ``alf_config.py`` in either case.
+For other changes to make for play, we need to modify the snapshot code if
+``--use_alf_snapshot`` is provided, and modify the current ALF repo otherwise.
+
+Advanced play by rendering
+--------------------------
+
+Besides playing with a snapshot, another advanced play case is to utilize the
+:mod:`.alf.summary.render` module. This module contains several helper functions
+that convert arrays and tensors to :class:`~.render.Image` objects for visualization
+on screen or in a video. Any :class:`~.render.Image` object, once put into
+``info`` of an ``AlgStep`` returned by an algorithm's ``predict_step()``, will
```

Comment:
Probably worth explaining "why `predict_step()`" for the new adopters, or at least remind them what does `predict_step()` do and when it is called. What do you think?

Reply:
Added several sentences for explanation.

Reply:
> Probably worth explaining "why `predict_step()`" for the new adopters, or at least remind them what does `predict_step()` do and when it is called. What do you think?

Added several sentences for explanation.